### PR TITLE
Cannot import local package collections

### PR DESCRIPTION
**Pull request by yim-lee**
Motivation:
Try passing a local FS path (file:///foo/bar) to the command `swift-package-collection describe`. It fails with `Error: invalidResponse`. This is because we try to fetch local file with HTTP client.

Modications:
If a collection URL points to local FS path, read the file contents instead of downloading it via HTTP GET.

Also improve error messages.

Result:
Local package collections can be imported.

### CODE REVIEW

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Comment by yim-lee** [comment]
@swift-ci please smoke test

File: Sources/PackageCollections/Model/Collection.swift [hunk 1]
```diff
@@ -98,6 +99,16 @@ extension PackageCollectionsModel {
     }
 }
 
+extension PackageCollectionsModel.CollectionSource {
+    /// Returns the source's absolute file system path, if its URL is of 'file' scheme.
+    var absolutePath: AbsolutePath? {
```
Comment (federicobucchi) [inline Sources/PackageCollections/Model/Collection.swift:104]: Can we use `let` ?

File: Sources/PackageCollections/Model/Collection.swift [hunk 2]
```diff
@@ -98,6 +99,16 @@ extension PackageCollectionsModel {
...
+        guard self.url.scheme?.lowercased() == "file" else {
+            return nil
+        }
+        return try? AbsolutePath(validating: self.url.path)
```
Comment (tomerd) [inline Sources/PackageCollections/Model/Collection.swift:108]: self.url.absoluteString

File: Sources/PackageCollections/PackageCollections+Validation.swift [hunk 3]
```diff
@@ -26,9 +26,15 @@ extension Model.CollectionSource {
...
-                appendMessage(.error("Non-local files not allowed: \(url.absoluteString)"))
+                appendMessage(.error("Scheme (\"\(scheme)\") not allowed: \(url.absoluteString). Must be one of \(allowedSchemes)."))
+            } else if scheme == "file" {
+                let absolutePath = self.absolutePath
+
+                if absolutePath == nil {
+                    appendMessage(.error("Invalid file path: \(self.url.path). It must be an absolute file system path."))
+                } else if let absolutePath = absolutePath, !localFileSystem.exists(absolutePath) {
+                    appendMessage(.error("Non-local files not allowed: \(self.url.path)"))
+                }
```
Comment (tomerd) [inline Sources/PackageCollections/PackageCollections+Validation.swift:37]: nice improvement

**Review by tomerd — APPROVED** [review]

File: Tests/PackageCollectionsTests/JSONPackageCollectionProviderTests.swift [hunk 4]
```diff
@@ -136,7 +120,10 @@ class JSONPackageCollectionProviderTests: XCTestCase {
             guard let internalError = (error as? MultipleErrors)?.errors.first else {
                 return XCTFail("invalid error \(error)")
             }
-            XCTAssertEqual(internalError as? ValidationError, ValidationError.other(message: "Schema not allowed: \(url.absoluteString)"))
+            guard let validationError = internalError as? ValidationError, case .other(let message) = validationError else {
+                return XCTFail("invalid error \(error)")
+            }
+            XCTAssertTrue(message.contains("Scheme (\"ftp\") not allowed: \(url.absoluteString)"))
         })
     }
```
Comment (tomerd) [inline Tests/PackageCollectionsTests/JSONPackageCollectionProviderTests.swift:128]: add tests for the new validation?

File: Sources/PackageCollections/Model/Collection.swift (hunk 2 above)
Comment (yim-lee) [reply Sources/PackageCollections/Model/Collection.swift:108]: `url.absoluteString` includes `file://` and `AbsolutePath` must start with `/` so the initialize will fail if we use it

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Pull request merged**